src/Microsoft.Dnx.Runtime.Internals/RuntimeEnvironmentExtensions.cs

```diff
@@ -3,12 +3,23 @@
 
 using System;
 using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
 using System.Text;
 
 namespace Microsoft.Dnx.Runtime
 {
-    internal static class RuntimeEnvironmentExtensions
+    internal static class RuntimeEnvironmentExtensions 
     {
+
+        private static Dictionary<Version, Version> _windowsVersionMapping = new Dictionary<Version, Version>()
```

Comment:
Also, I would avoid this dictionary completely and do the if else that was before because this code is on the startup hot path